src/lib/sql/sql_translator.cpp

```diff
@@ -654,6 +658,41 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_stored_table(
   return validated_stored_table_node;
 }
 
+std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_meta_table(
+    const std::string& name, const std::shared_ptr<SQLIdentifierResolver>& sql_identifier_resolver) {
+  AssertInput(MetaTableManager::is_meta_table_name(name), std::string{"Did not find a meta table with name "} + name);
+
+  // MetaTables are non-cacheable because they might contain information about the general system state
+  // that can change at any time
+  _cacheable = false;
+
+  const auto meta_table_name = name.substr(MetaTableManager::META_PREFIX.size());
+
+  // Meta tables are integrated in the LQP as static table nodes in order to avoid regeneration at every
+  // access in the pipeline afterwards.
+  std::shared_ptr<Table> meta_table;
+  if (_meta_tables->contains(meta_table_name)) {
+    meta_table = _meta_tables->at(meta_table_name);
+  } else {
+    meta_table = Hyrise::get().meta_table_manager.generate_table(meta_table_name);
+    (*_meta_tables)[meta_table_name] = meta_table;
+  }
+
+  const auto static_table_node = StaticTableNode::make(meta_table, false);
+  const auto validated_static_table_node = _validate_if_active(static_table_node);
```

Comment:
You might have discussed that already but why do we need validation for meta tables?

Reply:
you are right. this seems useless. 
but since the latest changes in the master allow entries of certain meta tables to be updated, inserted or deleted and these operators require the table to be validated, this validation step is necessary. 
An option would be to check for mutability of the meta table and validate accordingly, but i don't know if this really has an impact, since so far, meta tables tend to be very small.

Reply:
Correct, we do have inserts, updates, and deletes on meta tables now but the tables are still generated on the fly for each access. For example, updates do not really change a table but some internal data structure that is part of the meta table object. Therefore, they should never contain invisible rows.

Reply:
we should never need it. it was just required for all kinds of tables from the tests. removed it for meta tables in the tests